airbyte-integrations/connectors/source-pendo-python/source_pendo_python/streams.py

```diff
@@ -189,6 +190,82 @@ def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
         return "report"
 
 
+class ReportResult(PendoPythonStream):
+    json_schema = None  # Field to store dynamically built Airbyte Stream Schema
+    primary_key = "reportId"
+
+    def __init__(self, report: Mapping[str, Any], **kwargs):
+        super().__init__(**kwargs)
+        self.report = report
+        self.report_name = f"report_result_{report['id']}"
+
+    @property
+    def name(self):
+        return self.report_name
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"report/{self.report['id']}/results.json"
+
+    def parse_response(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+        for record in response.json():
+            yield self.transform(record=record)
+
+    def transform(self, record: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        record["reportId"] = self.report['id']
+        return record
+
+    # Method to infer schema types from JSON response
+    def infer_type(self, value: Any):
+        if isinstance(value, str):
+            return {"type": ["null", "string"]}
+        if isinstance(value, bool):
+            return {"type": ["null", "boolean"]}
+        if isinstance(value, float):
+            return {"type": ["null", "number"]}
+        if isinstance(value, int):
+            return {"type": ["null", "integer"]}
+        if isinstance(value, list):
+            return {"type": ["null", "array"]}
+        if isinstance(value, dict):
+            return {"type": ["null", "object"]}
```

Comment:
🧠  does Airbyte provide a helper we can reuse here?  What about date types?

Reply:
That was my first instinct. I took a look around and couldn't find an example for this sort of parsing in the repo, docs, or slack channel. I could have missed it or looked at the wrong places. I know this works for the Pendo data but if you have tips to find airbyte helpers I can replace it in a follow-up jira. As for dates, the Pendo data stores them as integers so that's how we need to provide them to the Airbyte schema.